cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -322,6 +330,59 @@ func (s *Server) Run(queue *Queue) {
 	}
 }
 
+// Update priority.json given new entries, and remove no longer tracked repos.
+// This doesn't simply write newPriorities because a transient getIndexOptions failure
+// would cause the associated repo to get deprioritized.
+func (s *Server) maybeUpdatePriorities(names []string, newPriorities map[string]float64) {
+	priorityPath := path.Join(s.IndexDir, "priority.json")
+	priorities := map[string]float64{}
+	buf, err := ioutil.ReadFile(priorityPath)
+	if err == nil {
+		err = json.Unmarshal(buf, &priorities)
+		if err != nil {
+			log.Printf("warning: error loading old priority.json, treating as empty: %v", err)
+		}
+	}
+
+	// maybe remove no-longer-tracked repos from the priorities list
+	if len(names) != len(priorities) {
+		set := make(map[string]struct{}, len(names))
+		for _, name := range names {
+			set[name] = struct{}{}
+		}
+		for name := range priorities {
+			if _, ok := set[name]; !ok {
+				delete(priorities, name)
+			}
+		}
+	}
+
+	for name, priority := range newPriorities {
+		priorities[name] = priority
+	}
+
+	newBuf, err := json.Marshal(priorities)
```

Comment:
maybe use `MarshalIndent` for more human readable json (and avoiding the need to append `\n`)